functions/src/converter.ts

```diff
@@ -0,0 +1,79 @@
+import * as yargs from 'yargs';
+
+//converter part 1
+
+const args = yargs.option('inputPath', {alias: 'i', demand: true, type: 'string'}).option('outputPath',
+               {alias: 'o', demand: true, type:'string'}).argv;
+const fs = require('fs');
+const rawJSON = fs.readFileSync(args.inputPath);
+const dataJSON = JSON.parse(rawJSON);
+convertToDBSchema(dataJSON);
+
+function convertToDBSchema(data: { vulnerabilities: any; ASB: any; published: any; }) : void {
+const SPLsArray : any[] = [];
+const found : string[] = [];
+ for(const vulnerabilities of data.vulnerabilities){
+    const patchLevel = vulnerabilities.patch_level;  
+    if (found.indexOf(patchLevel) === -1 && patchLevel !== undefined) {
+        found.push(patchLevel);
+        SPLsArray.push(patchLevel);
+    }
+}
+const SPLs = {
+    'SPLs': SPLsArray
+ };
+
+let ASBs = null;
+if (data.ASB === undefined){
+    const newDate = data.vulnerabilities[0].ASB;
+    console.log(newDate);
+    ASBs = { [newDate] : SPLs };
+}
+else {
+    ASBs = { [data.ASB] : SPLs };
+}
+
+const Bulletin_SPLs = {'ASBs': ASBs};
+
+const publishDate = data.published;
+const result : Record<string, object> = {};
+for(const spl of SPLsArray){
+    const cveIDsArray : string[] = [];
+    const splDetails = {
+        'Publish_Date': publishDate,
+        'CVE_IDs': cveIDsArray
+     };
+     for(const vulnerabilities of data.vulnerabilities){
+        const patchLevel = vulnerabilities.patch_level;
+        if(patchLevel === spl){
+            const cveID = vulnerabilities.CVE;
+            if (cveID !== undefined){
+                splDetails.CVE_IDs.push(cveID);
+                }
+            }
+        result[spl] = splDetails;
+      }
+}
+
+const SPL_CVE_IDs = {'SPLs' : result};
+const output = {SPL_CVE_IDs, Bulletin_SPLs};
+
+//TODO
+// const AOSP_VERSION_CVE_IDs 
+// const AOSP_VERSION_ASB_CVE_IDs 
+// const CVEs 
+
+fs.writeFileSync(args.outputPath, JSON.stringify(output));
+console.log("Completed conversion!");
+}
+
+/*To run this script:
```

Comment:
I'd suggest creating a separate script directory somewhere, and then having individual directories for each script + a README.md with these steps